core/templates/dev/head/editor/feedback_tab/ThreadDataService.js

```diff
@@ -167,8 +167,8 @@ oppia.factory('threadDataService', [
         payload.commit_message = commitMsg;
       }
       _openThreadsCount -= 1;
-      $http.put(_SUGGESTION_ACTION_HANDLER_URL + threadId, payload).success(
-        onSuccess).error(function() {
+      $http.put(_SUGGESTION_ACTION_HANDLER_URL + threadId, payload).then(
+        onSuccess, function() {
```

Comment:
Sorry, I don't understand your response, so let me clarify my question. When this $http.put() is called, what are the possibilities for onSuccess? Here, onSuccess is a placeholder for the success promise, so, in all those possibilities, we should check that the first argument is a 'response' object (and not a 'data' object).

(Again, the correct behaviour should be verifiable by manual testing.)